auction-server/src/opportunity/service/remove_invalid_or_expired_opportunities.rs

```diff
@@ -63,3 +63,209 @@ impl Service {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use {
+        crate::{
+            api::ws::UpdateEvent,
+            kernel::{
+                rpc_client_svm_tester::RpcClientSvmTester,
+                test_utils::DEFAULT_CHAIN_ID,
+            },
+            opportunity::{
+                entities::{
+                    FeeToken,
+                    OpportunitySvm,
+                    OpportunitySvmProgram,
+                    OpportunitySvmProgramSwap,
+                    TokenAccountInitializationConfig,
+                    TokenAccountInitializationConfigs,
+                    TokenAmountSvm,
+                },
+                repository::MockDatabase,
+                service::Service,
+            },
+        },
+        ethers::{
+            prelude::Bytes,
+            utils::hex::FromHex,
+        },
+        express_relay_api_types::opportunity::{
+            OpportunityDelete,
+            OpportunityDeleteSvm,
+            OpportunityDeleteV1Svm,
+            ProgramSvm,
+        },
+        solana_sdk::pubkey::Pubkey,
+        time::{
+            Duration,
+            OffsetDateTime,
+        },
+        uuid::Uuid,
+    };
+
+    fn make_test_opportunity(
+        id: u128,
+        creation_time: OffsetDateTime,
+        refresh_time: OffsetDateTime,
+    ) -> OpportunitySvm {
+        OpportunitySvm {
+            id: Uuid::from_u128(id),
+            permission_key: Bytes::from_hex("0xdeadbeef").unwrap(),
+            chain_id: DEFAULT_CHAIN_ID.to_string(),
+            sell_tokens: vec![TokenAmountSvm {
+                token:  Pubkey::new_unique(),
+                amount: 2,
+            }],
+            buy_tokens: vec![TokenAmountSvm {
+                token:  Pubkey::new_unique(),
+                amount: 1,
+            }],
+            creation_time,
+            refresh_time,
+            router: Pubkey::new_unique(),
+            permission_account: Pubkey::new_unique(),
+            program: OpportunitySvmProgram::Swap(OpportunitySvmProgramSwap {
```

Comment:
can you use `OpportunitySvmProgramSwap::default_test_with_user_wallet_address` here?